eox_tenant/tenant_wise/proxies.py

```diff
@@ -150,18 +150,20 @@ def __get_value_for_org(cls, org, val_name, default=None):
         TenantConfig or Microsite model.
         """
         cache_key = "org-value-{}-{}".format(org, val_name)
+        if default is not None:
+            cache_key = "{}-{}".format(cache_key, default)
```

Comment:
The `default` values are mostly fetched from django settings, which for most part are configured in the tenant configs. So with the current implementation, we are essentially, re-fetching the same value from tenant config. To avoid this, what if we simply do the following and remove the other changes in `model.py` ?

```suggestion
        if default is not None:
            return default
```
Would there be any risk, edge cases with this ?

Reply:
@kaustavb12 That is a nice idea! But if an organization is not present in the current tenant config's `org_filter` this method will fail as it will still return value from the current config/settings.

Reply:
Ahh.. yes. That's a good point.